File: 프로그래머스/lv2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    process = []
    cnt = 1
    for i in range(len(priorities)):
        process.append((i,priorities[i]))
    while len(process)>0:
        max = 0
        for i in range(len(process)):
            if max < process[i][1]:
                max = process[i][1]
        if process[0][1] == max : 
            if process[0][0] == location:
                return cnt
            else:
                del process[0]
                cnt+=1
        else:
            process.append((process[0][0],process[0][1]))
            del process[0] 

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let arr = []
    let ans = 1
    for(let i=0;i<priorities.length;i++){
        arr.push([i,priorities[i]])
    }
    priorities.sort((a,b)=>b-a)
    while(arr.length>0){
        if(priorities[0]===arr[0][1] && arr[0][0]===location) return ans
        else if(priorities[0]===arr[0][1]) {
            arr.shift()
            priorities.shift()
            ans++
        } else {
            arr.push(arr.shift())
        }
    }
}